frontend/src/hooks/useLogout.ts

```diff
@@ -1,36 +1,34 @@
-import React, { useContext } from "react";
+import { useContext } from "react";
+import { useHistory } from "react-router-dom";
 import { gql, useMutation } from "@apollo/client";
-
-import authAPIClient from "../../APIClients/AuthAPIClient";
-import AuthContext from "../../contexts/AuthContext";
+import authAPIClient from "../APIClients/AuthAPIClient";
+import AuthContext from "../contexts/AuthContext";
 
 const LOGOUT = gql`
   mutation Logout($userId: ID!) {
     logout(userId: $userId)
   }
 `;
 
-const Logout = (): React.ReactElement => {
+const useLogout = (): (() => Promise<void>) => {
   const { authenticatedUser, setAuthenticatedUser } = useContext(AuthContext);
 
+  const history = useHistory();
+
   const [logout] = useMutation<{ logout: null }>(LOGOUT);
 
-  const onLogOutClick = async () => {
+  const tryLogout = async () => {
     try {
       await authAPIClient.logout(String(authenticatedUser?.id), logout);
+      await tryLogout();
```

Comment:
So this will call itself until it fails, and only then will it jump into the catch handler and exit the function? Like, we are intentionally failing/expecting an error, because that means we aren't authenticated anymore. This seems like an interesting/counterintuitive way to go about it. What happens if the authAPIClient.logout fails an we continue to be logged in? Will we be stuck in an infinite recursive loop?

Reply:
Yeah, exactly; that's how the existing logic works at least. I considered limiting it to a certain number of tries to log out, but not sure if that's worth it. It should usually fail in this case. Additionally I could add an additional check on the return value of `authAPIClient.logout`.